### ISSUE TITLE
false SC2218

### ISSUE THREAD

**Issue by pan64** [koalaman/shellcheck#3421]
#### For bugs with existing features

- Rule Id (if any, e.g. SC2218): SC2218
- My shellcheck version (`shellcheck --version` or "online"): 0.11.0
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [ ] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit - cannot be checked, since it needs sourcing

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
# shellcheck source=path/log.sh                                                                                                                                       
source "${VAR}/path/log.sh" 

log_info "message"
```

#### Here's what shellcheck currently says:
^-- SC2218 (error): This function is only defined later. Move the definition up.

#### Here's what I wanted or expected to see:
As it is defined in the sourced log.sh it should be accepted without issues.

(VAR is actually the current dir)
0.10.0 worked.

**Comment by brother** [comment]
Actually not able to repeat this 


PS replacing "source" with full stop to silence that specific warning...

**Comment by pan64** [comment]
Ok, a little bit more difficult to reproduce, but it is here:

```
/tmp/lib.sh:
function my_func() {
    :
}
```

```
/tmp/test2.sh:
# shellcheck source=/tmp/lib.sh
source /tmp/lib.sh
```

```
/tmp/test1.sh:
#!/bin/bash
# shellcheck source=/tmp/lib.sh
source /tmp/lib.sh
my_func "test"
# shellcheck source=/tmp/test2.sh
source /tmp/test2.sh
```


Error message:
in /tmp/test1.sh line 7:
my_func "test"
^------------^ SC2218 (error): This function is only defined later. Move the definition up.

**Comment by SimeonEhrig** [comment]
Can confirm. I have the same problem. A function should be not defined, which is defined in a sourced bash script. But only for single function and after I changed unrelated code at another location.
